### PR TITLE
#169258621 User Registration

### PR DESCRIPTION
**Pull request by william000000**
#### What does this PR do?
Users should be able to sign up and receive descriptive signup/Registration validation errors or success.
#### Description of Task to be completed?
- Add validations
- Add tests
- Create user tables
- Use JSON Web token
- Swagger documentation 
#### How should this be manually tested?
- You have to clone the repo `git clone https://github.com/andela/team-odd-bn-backend.git`
- Then After the repo cloned `cd project-folder` and run `npm i` to install all dependencies
- Go to my branch `git checkout ft-signup-user-169258630`
- Run app by `npm run dev-start`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169258621](https://www.pivotaltracker.com/story/show/169258630)
#### Screenshots (if appropriate)
image
image

#### Questions:
N/A

### CODE REVIEW

File: src/controllers/UserController.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        lastName,
+        email,
+        password: hashedPassword
+      });
+      const {
+        password,
+        ...data
+      } = newUser.dataValues;
+      const token = tokenGenerator.signToken(data);
+      data.token = token;
```
Comment (julietezekwe) [inline src/controllers/UserController.js:41]: Please take out commented code

File: src/routes/api/authRoute.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import express from 'express';
+import userController from '../../controllers/UserController';
+import isUserExist from '../../middlewares/FindUsers';
+import validateUser from '../../middlewares/ValidateUser';
+
+const authRouter = express.Router();
+
```
Comment (julietezekwe) [inline src/routes/api/authRoute.js:7]: Please refactor the route to have just `/signup` and add auth to the route index

File: src/routes/api/index.js [hunk 3]
```diff
@@ -1,18 +1,7 @@
-const router = require("express").Router();
+import express from 'express';
+import users from './authRoute';
 
-router.use("/", require("./users"));
+const router = express.Router();
+router.use('/', users);
```
Comment (julietezekwe) [inline src/routes/api/index.js]: Import as `authRouter` and use /auth instead

File: src/routes/api/authRoute.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import express from 'express';
+import userController from '../../controllers/UserController';
```
Comment (julietezekwe) [inline src/routes/api/authRoute.js]: Use PascalCase when importing a class

**Review by julietezekwe — CHANGES_REQUESTED** [review]
Great job you have here @william000000, Kindly address the comments and revert. 💪🏼

**Review by hezronkimutai — COMMENTED** [review]
Nicely done, @willy, check the comments i left

File: src/routes/api/index.js [hunk 5]
```diff
@@ -1,18 +1,7 @@
-const router = require("express").Router();
+import express from 'express';
+import users from './authRoute';
```
Comment (julietezekwe) [inline src/routes/api/index.js]: Please change this to import authRoute from './authRoute';

File: src/routes/api/index.js [hunk 6]
```diff
@@ -1,18 +1,7 @@
...
 
-router.use("/", require("./users"));
+const router = express.Router();
+router.use('/auth', users);
```
Comment (julietezekwe) [inline src/routes/api/index.js]: Then change this users here to authRoute for name convention

**Review by julietezekwe — CHANGES_REQUESTED** [review]

File: src/routes/api/index.js (hunk 6 above)
Comment (william000000) [reply src/routes/api/index.js]: Corrected

**Review by julietezekwe — APPROVED** [review]

**Review by hezronkimutai — APPROVED** [review]

**Review by NiyoEric — APPROVED** [review]
The project is well structured. Well done!

**Comment by victkarangwa** [comment]
Nice work @william000000

**Review by victkarangwa — COMMENTED** [review]
Good to go!

**Pull request merged**